src/CMakeLists.txt

```diff
@@ -1,106 +1,42 @@
-
-if (NOT WIN32 AND (CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "GNU"))
-  set(xournalpp_LDFLAGS ${xournalpp_LDFLAGS} -rdynamic)
-endif ()
-
-option (DEBUG_COMPILE "Pass -Wall to CXX_FLAGS" OFF)
-mark_as_advanced (FORCE DEBUG_COMPILE)
-
-if (DEBUG_COMPILE)
-  add_definitions (-Wall)
-endif (DEBUG_COMPILE)
-
-if (DEV_ENABLE_GCOV)
-  add_definitions (--coverage)
-endif (DEV_ENABLE_GCOV)
-
-# Used to compile xournalpp and xournalpp-test
-set (xournalpp_LDFLAGS
-  util
-  ${xournalpp_LDFLAGS}
-)
-
-## Build util lib ##
-add_subdirectory (util)
-
-## Sources ##
-
-# These dirs are xournalpp only so it's safe to add then recursively
-unset (xournalpp_SOURCES_RECURSE)
-file (GLOB_RECURSE xournalpp_SOURCES_RECURSE
-  control/*.cpp
-  enums/*.cpp
-  gui/*.cpp
-  io/*.cpp
-  model/*.cpp
-  plugin/*.cpp
-  undo/*.cpp
-  view/*.cpp
-)
-
-# Here GLOB_RECURSE wouldn't be safe
-unset (xournalpp_SOURCES)
-file (GLOB xournalpp_SOURCES
-  pdf/base/*.cpp
-)
-
-# Concatenate SOURCES lists
-set (xournalpp_SOURCES ${xournalpp_SOURCES_RECURSE} ${xournalpp_SOURCES})
-unset (xournalpp_SOURCES_RECURSE)
-
-file (GLOB xournalpp_SOURCES_RECURSE
-  pdf/popplerapi/*.cpp
-)
-
-# Concatenate SOURCES lists
-set (xournalpp_SOURCES ${xournalpp_SOURCES_RECURSE} ${xournalpp_SOURCES})
-unset (xournalpp_SOURCES_RECURSE)
-
-## Core library ##
-
-# Used for xournalpp and xournalpp-test
-add_library (xournalpp-core OBJECT ${xournalpp_SOURCES})
-add_dependencies (xournalpp-core util)
-target_compile_features (xournalpp-core PUBLIC ${PROJECT_CXX_FEATURES})
-# Can't use target_link_directories since we require CMake >= 3.10
-target_include_directories (xournalpp-core PRIVATE $<TARGET_PROPERTY:util,INTERFACE_INCLUDE_DIRECTORIES>)
-
-## xournalpp main program ##
-add_executable (xournalpp
-  $<TARGET_OBJECTS:xournalpp-core>
-  Xournalpp.cpp
-)
-
-add_dependencies (xournalpp
-  xournalpp-core
-  util
-)
-
-target_link_libraries (xournalpp ${xournalpp_LDFLAGS})
-
-install (TARGETS xournalpp
-  RUNTIME DESTINATION bin
-  COMPONENT xournalpp
-)
+cmake_minimum_required(VERSION 3.11)
+
+# Todo: add to target by generator expression
+#if (DEV_ENABLE_GCOV)
+#    add_definitions(--coverage)
+#endif (DEV_ENABLE_GCOV)
+
+##############################
+# xoj default option target
+##############################
+add_library(defaults INTERFACE)
+target_compile_options(defaults INTERFACE -Wall -Wreturn-type -Wuninitialized -Wunused-value -Wunused-variable)
+target_include_directories(defaults INTERFACE ${PROJECT_BINARY_DIR}/src)
+add_library(xoj::defaults ALIAS defaults)
+
+##############################
+# util
+##############################
+
+add_subdirectory(util)
+##############################
+# xournalpp-core
+##############################
+add_subdirectory(core)
+
+##############################
+# xournalpp main program
+##############################
+
+add_executable(xournalpp core/Xournalpp.cpp)
```

Comment:
If the file is in the core directory it should also be part of the core target. Either refactor this back out to the main directory or move this call to the cmake script within the directory and use [this property](https://cmake.org/cmake/help/latest/prop_tgt/RUNTIME_OUTPUT_DIRECTORY.html#prop_tgt:RUNTIME_OUTPUT_DIRECTORY) to build the binary in the correct directory.

We should also consider using `WIN32` and `MACOSX_BUNDLE` in this call to improve support within Windows and OSX.

Reply:
moved it to a dir named `exe/`